src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -27,8 +31,131 @@ type PaginatedData = {
+  const transactionMap: { [txHash: string]: TokenTransaction } = {}
+
+  for (const tx of transactions) {
+    transactionMap[tx.transactionHash] = tx
+  }
+
+  return Object.values(transactionMap)
+}
+
+/**
+ * If the timestamps are the same, most likely one of the transactions is an approval.
+ * On the feed we want to show the approval first.
+ */
+function sortTransactions(transactions: TokenTransaction[]): TokenTransaction[] {
+  return transactions.sort((a, b) => {
+    const diff = b.timestamp - a.timestamp
+
+    if (diff === 0) {
+      return a.__typename === 'TokenApproval' ? 1 : b.__typename === 'TokenApproval' ? -1 : 0
+    }
+
+    return diff
+  })
+}
+
+/**
+ * Every page of paginated data includes a limited amount of transactions within a certain period.
+ * In standByTransactions we might have transactions from months ago. Whenever we load a new page
+ * we only want to add those stand by transactions that are within the time period of the new page.
+ * Otherwise, if we merge all the stand by transactins into the page it will cause more late transactions
+ * that were already merged to be removed from the top of the list and move them to the bottom.
+ * This will cause the screen to "shift", which we're trying to avoid.
+ */
+function mergeStandByTransactionsInRange(
+  transactions: TokenTransaction[],
+  standBy: TokenTransaction[]
+) {
+  if (transactions.length === 0) return []
+
+  const allowedNetworks = getAllowedNetworksForTransfers()
+  const max = transactions[0].timestamp
+  const min = transactions.at(-1)!.timestamp
+
+  const standByInRange = standBy.filter((tx) => tx.timestamp >= min && tx.timestamp <= max)
+  const deduplicatedTransactions = deduplicateTransactions([...transactions, ...standByInRange])
+  const transactionsFromAllowedNetworks = deduplicatedTransactions.filter((tx) =>
+    allowedNetworks.includes(tx.networkId)
+  )
+
+  return transactionsFromAllowedNetworks
+}
+
+/**
+ * Current implementation of allStandbyTransactionsSelector contains function
+ * getSupportedNetworkIdsForApprovalTxsInHomefeed in its selectors list which triggers a lot of
+ * unnecessary re-renders. This can be avoided if we join it's result in a string and memoize it,
+ * similar to how it was done with useAllowedNetworkIdsForTransfers hook from queryHelpers.ts
+ *
+ * Not using existing selectors for pending/confirmed stand by transaction only cause they are
+ * dependant on the un-memoized standbyTransactionsSelector selector which will break the new
+ * pagination flow.
+ *
+ * Implementation of pending is identical to pendingStandbyTransactionsSelector.
+ * Implementation of confirmed is identical to confirmedStandbyTransactionsSelector.
+ */
+function useStandByTransactions() {
+  const standByTransactions = useSelector(allStandbyTransactionsSelector)
+  const allowedNetworkForTransfers = useAllowedNetworksForTransfers()
+  const supportedNetworksForApproval = useSupportedNetworksForApproval()
```

Comment:
I think we can remove this now. All networks are supposed to be supported for approvals with Zerion.